src/components/Block.jsx

```diff
@@ -20,20 +21,31 @@ module.exports = React.createClass({
     return app.refine('blockTypes').find(i => i.id === block.type)
   },
 
+  setMenuItems(component) {
+    if ('getMenuItems' in component) {
+      this.setState({ extraMenuItems: component.getMenuItems() })
+    }
+  },
+
+  setComponentRef(component) {
+    this.setMenuItems(component)
+    return 'block'
+  },
```

Comment:
Slightly confused about this — is a function passed to ref automatically called with `component` as an arg? I did not know this behavior.